src/poetry/factory.py

```diff
@@ -205,23 +212,50 @@ def create_legacy_repository(
         )
 
     @classmethod
-    def create_pyproject_from_package(cls, package: ProjectPackage, path: Path) -> None:
+    def create_pyproject_from_package(
+        cls, package: Package, path: Path | None = None
+    ) -> TOMLDocument:
         import tomlkit
 
-        from poetry.layouts.layout import POETRY_DEFAULT
-
-        pyproject: dict[str, Any] = tomlkit.loads(POETRY_DEFAULT)
-        content = pyproject["tool"]["poetry"]
+        pyproject = tomlkit.document()
```

Comment:
This is losing typing fixes.  Please consider running mypy at least locally in the poetry virtual environment - and also merging or otherwise resolving #5279.

```
src/poetry/factory.py:222: error: Item "Item" of "Union[Item, Container]" has no attribute "_is_super_table"  [union-attr]
src/poetry/factory.py:222: error: Item "Container" of "Union[Item, Container]" has no attribute "_is_super_table"  [union-attr]
src/poetry/factory.py:223: error: Unsupported target for indexed assignment ("Union[Item, Container]")  [index]
src/poetry/factory.py:253: error: "Item" has no attribute "append"  [attr-defined]
src/poetry/factory.py:286: error: "Item" has no attribute "_is_super_table"  [attr-defined]
src/poetry/factory.py:288: error: Unsupported right operand type for in ("Item")  [operator]
src/poetry/factory.py:289: error: Unsupported target for indexed assignment ("Item")  [index]
src/poetry/factory.py:290: error: Value of type "Item" is not indexable  [index]
src/poetry/factory.py:292: error: Value of type "Item" is not indexable  [index]
src/poetry/factory.py:293: error: Value of type "Item" is not indexable  [index]
src/poetry/factory.py:295: error: Value of type "Item" is not indexable  [index]
```